include/dpp/message.h

```diff
@@ -341,6 +342,16 @@ class DPP_EXPORT component : public json_interface<component> {
 	 */
 	component& set_label(const std::string &label);
 
+	/**
+	 * @brief Set the value of the modal component.
+	 * For action rows, this field is ignored. Setting the
+	 * value will auto-set the type to dpp::cot_text.
+	 *
+	 * @param value Value text to set. It will be truncated to the maximum length of 4000 UTF-8 characters.
+	 * @return component& Reference to self
+	 */
+	component& set_value(const std::string &val);
```

Comment:
the value is not just string, see the value member, there should be overloaded versions for int64 and double too if you're adding this

Reply:
But, the API specifies that the json only accepts a string.

Reply:
where though? this is in dpp component which is used in a few places?

Reply:
This is just for setting the default value of the modal, not anywhere else. [source](https://discord.com/developers/docs/interactions/message-components#text-inputs).

Reply:
no  a modal can only use text inputs but there are other kinds of input, and elsewhere a text input component may contain text, integers or double:
https://discord.com/developers/docs/interactions/message-components#component-object-component-types